python/dgl/graphbolt/itemset.py

```diff
@@ -255,6 +267,44 @@ def __iter__(self) -> Iterator:
     def __len__(self) -> int:
         return sum(len(itemset) for itemset in self._itemsets.values())
 
+    def __getitem__(self, idx: Union[int, slice, Iterable]) -> Dict[str, Tuple]:
+        try:
+            len(self)
+        except TypeError:
+            raise TypeError(
+                f"{type(self).__name__} instance doesn't support indexing."
+            )
+        offsets = [len(itemset) for itemset in self._itemsets.values()]
+        total_num = sum(offsets)
+        if isinstance(idx, int):
+            if idx < 0:
+                idx += total_num
+            assert idx < total_num, (
```

Comment:
should we assert idx >= 0?

Reply:
no as negative index is converted to positive ones. and I'd like to support negative indexing

Reply:
I meant after idx += total_num, the idx should be non-negative.

Reply:
right. should be added.